cloudflare.go

```diff
@@ -159,31 +159,31 @@ func (api *API) makeRequestWithAuthType(ctx context.Context, method, uri string,
 }
 
 func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, uri string, params interface{}, authType int, headers http.Header) ([]byte, error) {
-	// Replace nil with a JSON object if needed
-	var jsonBody []byte
 	var err error
+	var reqBody io.Reader
```

Comment:
I believe that by moving the io.Reader setup outside of the retry loop you run the risk of retries being broken, since the first attempt may consume the reader and subsequent attempts will then have no request body to read from.

If that's the case it may make more sense to consume `params` into a byte slice outside of the retry loop and create a reader out of it within the retry loop, similar to the original implementation.

If params *is* an io.Reader, maybe just ioutil.ReadAll to read it into the byte slice.

Reply:
Ok I have made the change as suggested.